rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,921 @@
+  root: {
+    /* styles */
+  },
+  foo: {
+    /* styles */
+  },
+  bar: {
+    /* styles */
+  },
+});
+
+function Component(props) {
+  const classes = useClasses();
+  const className = mergeClasses(
+    classes.root,
+    props.foo && classes.foo /* styles specific for "foo" */,
+    props.bar && classes.bar /* styles specific for "bar" */,
+  );
+
+  /* --- */
+}
+```
+
+### Order of arguments determines results
+
+Unlike native CSS, the output of mergeClasses() is affected by the order of the classes passed in, allowing for control over priority of style overrides.
+
+```js
+import { mergeClasses, makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  blue: { color: 'blue' },
+  red: { color: 'red' },
+});
+
+function Component(props) {
+  // ℹ️ Order of arguments determines the results
+
+  const redClassName = mergeClasses(classes.blue, classes.red);
+  // 👆 { color: 'red' }
+  const blueClassName = mergeClasses(classes.red, classes.blue);
+  // 👆 { color: 'blue' }
+
+  /* --- */
+}
+```
+
+### ⚠️ Only combine classes with `mergeClasses`
+
+It is not possible to simply concatenate classes returned by `useClasses()` hooks. Always use `mergeClasses()` to merge classes as results of concatenation can contain duplicated classes and lead to non-deterministic results.
+
+```js
+import { makeStyles, mergeClasses } from '@griffel/react';
+
+const useClasses = makeStyles({
+  rootA: { display: 'flex' },
+  rootB: { display: 'grid' },
+});
+
+function App(props) {
+  const classes = useClasses();
+
+  // ✅ Returns "class-display-grid"
+  const correctClasses = mergeClasses(classes.rootA, classes.rootB);
+  // 🔴 Never concatenate class strings, returns "class-display-flex class-display-grid"
+  const wrongClasses = classes.rootA + ' ' + classes.rootB;
+}
+```
+
+## `makeResetStyles`
+
+This API works similarly to `makeStyles` and is used to styles as a single monolithic class to avoid the "CSS rules explosion" problem.
+
+```js
+import { makeResetStyles } from '@griffel/react';
+
+const useBaseClassname = makeResetStyles({
+  padding: '4px',
+  margin: '4px',
+});
+```
+
+> ⚠️ **Note:** Only one class generated by `makeResetStyles()` can be applied to an element. Otherwise, behavior will be non-deterministic since styles are not merged and deduplicated, the results will depend on order of insertion.
+
+### Hybrid approach (Using `makeStyles` and `makeResetStyles` together)
+
+We recommend using [`makeResetStyles` API][griffel-make-reset-styles] to define the base styles for a component and use `makeStyles` to override or enhance the base styles at runtime:
+
+```jsx
+import { makeStyles, makeResetStyles, mergeClasses, shorthands } from '@griffel/react';
+
+const useBaseClassname = makeResetStyles({
+  ':hover': {
+    ...shorthands.padding('4px'),
+    /* other styles */
+  },
+  '::before': {
+    display: 'block',
+    content: "' '",
+  },
+  '@media (forced-colors: active)': {
+    ...shorthands.padding('4px'),
```

Comment:
Is expanding shorthand in makeResetStyles a wasted work?
```suggestion
    padding: '4px',
```

Reply:
> Is expanding shorthand in makeResetStyles a wasted work?

Indeed, thanks 👍